docs/functionaldatatypes/validation.md

```diff
@@ -0,0 +1,286 @@
+
+If you have a whole collection of values you want to validate you can use the `collectAllPar` operator to validate them all at the same time.
+
+### Chaining Validations
+
+The second way we might want to combine `Validation` values is by returning another `Validation` value based on the result of the first validation value. For example, we might first try to parse a string into an integer and then validate that the integer is within a reasonable range.
+
+```scala mdoc:reset
+import zio.prelude.Validation
+
+def validateInt(s: String): Validation[String, Int] =
+  Validation(s.toInt).mapError(_.getMessage)
+
+def validateAge(age: Int): Validation[String, Int] =
+  Validation.fromPredicateWith(s"Age $age was less than zero")(age)(_ >= 0)
+
+def validateAgeString(s: String): Validation[String, Int] =
+  for {
+    n <- validateInt(s)
+    _ <- validateAge(n)
+  } yield n
+```
+
+Here the second validation depended on the first one because if we could not parse the string into an integer then we would not even have an integer to validate. Just like with other data types we can express this with the `flatMap` operator and use a _for comprehension_ to chain multiple validations together.
+
+Note that when we chain validations like this we only do the second validation if the first one is successful so we will never see both errors here. If all we are doing is chaining then we don't actually need `Validation` and could just use `Either`.
+
+However, frequently we want to do some validations "in parallel" and other "sequentially" and having `flatMap` on `Validation` makes it very easy for us to do this. Note that when we say in parallel here no actual parallelism is going on here, we just mean that we are performing both validations even if one of them fails.
+
+To see this, let's expand on our example of validating a `Person` data type from above. Now we will validate the person based on two `String` inputs using the logic we implemented above.
+
+```scala
+def validatePerson(name: String, age: String): Validation[String, Person] =
+  Validation.validateWith(validateName(name), validateAgeString(age))(Person)
+```
+
+Now we are combining "parallel" and "sequential" validations.
+
+Validating the age will proceed by validating that the age can be parsed into an `Int` and then that the age is not negative. This will return either a validated age or a validation error if either of these steps failed.
+
+We are also validating that the name is not empty. The final result will be either a validated person or a validation error containing both any errors that occurred in validating the name and age.
+
+By combining operators like `validateWith` and `flatMap` you can build complex validation logic that handles any of your problems in this or related domains.
+
+## Extracting Validation Values
```

Comment:
pedantic (API not doc): any thought of having a type check such as `isValid` predicate (and not doing something silly like `toOption.nonEmpty`)? It feels like it was left out intentionally somehow but `Either, Try` have them.

Reply:
I think it would be a good addition. Will add in a separate PR.